ccf_transaction_processor/CMakeLists.txt

```diff
@@ -19,12 +19,17 @@ project(pdoenc LANGUAGES C CXX)
 include(${CCF_DIR}/cmake/ccf_app.cmake)
 
 find_package(ccf REQUIRED)
+SET(SGX_MODE "$ENV{SGX_MODE}")
 
 # Add the PDO transaction processor target
 add_ccf_app(pdoenc
   SRCS transaction_processor/pdo_tp.cpp transaction_processor/verify_signatures.cpp
   INCLUDE_DIRS ${CCF_DIR}/include/ccf ${CCF_DIR}/include/3rdparty ${CCF_DIR}/include}
 )
 
-install(TARGETS pdoenc.virtual DESTINATION lib)
+IF(${SGX_MODE} MATCHES "HW")
+    message("SGX_MODE is HW")
+    TARGET_COMPILE_DEFINITIONS(pdoenc.virtual PRIVATE "CHECK_ATTESTATION=1")
+ENDIF()
 
+install(TARGETS pdoenc.virtual DESTINATION lib)
```

Comment:
would you handle environment variables & configuration consistently with the CMakefiles in the rest of the project? Check the definition, set the value, string equals (rather than matches). Consistently upper case CMake functions. Etc.

Reply:
thank you, I have capitalized the commands. IMHO, $PDO_SOURCE_ROOT/build/cmake/SGX.cmake is an overkill for CCF TP installation. Currently we rely entirely on ccf base for cmake definitions. the ccf base also includes definitions that are needed to deploy the TP in HW mode. The TP installation does not need PDO client/server installation. SGX_MODE is the only env variable that we need. and I dont even know if build/cmake might conflict with ccfbase/cmake.

Reply:
I'm not sure i agree with the "overkill" comment. It seems like you could drop the extra docker file if you can build with the right set of sgx libraries so you don't need the separate docker.